tests/kernel/workq/work/src/main.c

```diff
@@ -1302,6 +1302,8 @@ static bool try_queue_no_yield(struct k_work_q *wq)
 
 	zassert_equal(k_sem_take(&sync_sem, K_NO_WAIT), -EBUSY);
 
+	k_sem_reset(&sync_sem);
```

Comment:
> Just remove.  This doesn't look correct, and the change to the way
rescheduling works in zync is tripping over this.

Remove what? is the commit message outdated?

Reply:
Yeah, this looks like a mis-split.  Will track down where the right code went.